baybe/searchspace/core.py

```diff
@@ -303,23 +304,69 @@ def estimate_product_space_size(parameters: Iterable[Parameter]) -> MemorySize:
 
     def transform(
         self,
-        data: pd.DataFrame,
+        df: pd.DataFrame | None = None,
+        /,
+        *,
+        allow_missing: bool = False,
+        allow_extra: bool | None = None,
+        data: pd.DataFrame | None = None,
     ) -> pd.DataFrame:
-        """Transform data from experimental to computational representation.
-
-        This function can e.g. be used to transform data obtained from measurements.
-        Continuous parameters are not transformed but included.
+        """Transform parameters from experimental to computational representation.
 
         Args:
-            data: The data to be transformed. Must contain all specified parameters, can
-                contain more columns.
+            df: The dataframe to be transformed. The allowed columns of the dataframe
+                are dictated by the `allow_missing` and `allow_extra` flags.
+            allow_missing: If ``False``, each parameter of the space must have one
+                corresponding column in the given dataframe. If ``True``, the dataframe
+                may contain only a subset of parameter columns.
+            allow_extra: If ``False``, every column present in the dataframe must
```

Comment:
It is not clear what the choice of `None` has for an influence. If I understand it correctly, this is mainly to enable the deprecation mechanism - still, I think this should be mentioned here.

Reply:
Also, a comment like "Those additional columns will be ignored" might be helpful (although that should be kind of clear....)

Reply:
Both done